I created this one because semantically it's a separate issue from the 2 algorithm PRs. I was thinking you can merge this PR first, then the other ones if approved.

And thanks, good catch. Encountering this problem with `nosetests --with-coverage` though. I tried googling for 10min and reading docs but didn't find a meaningful solution. Have you run into this before?

```
$ nosetests --with-coverage
........................
Name    Stmts   Miss Branch BrPart  Cover
-----------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda3/envs/mdsa/bin/nosetests", line 6, in <module>
    sys.exit(nose.run_exit())
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/unittest/main.py", line 95, in __init__
    self.runTests()
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/core.py", line 66, in run
    result.printErrors()
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/result.py", line 110, in printErrors
    self.config.plugins.report(self.stream)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/plugins/manager.py", line 99, in __call__
    return self.call(*arg, **kw)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/plugins/manager.py", line 167, in simple
    result = meth(*arg, **kw)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/nose/plugins/cover.py", line 190, in report
    self.coverInstance.report(modules, file=stream)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/coverage/control.py", line 985, in report
    return reporter.report(morfs, outfile=file)
  File "/home/travis/miniconda3/envs/mdsa/lib/python2.7/site-packages/coverage/summary.py", line 140, in report
    raise CoverageException("No data to report.")

coverage.misc.CoverageException: No data to report.
The command "nosetests --with-coverage" exited with 1.
```

--- a/.gitignore
+++ b/.gitignore
@@ -41,6 +41,9 @@ pip-log.txt
 *.o
 *.so
 
+# coveralls
+.coverage
+
 # unwanted files
 .ipynb_checkpoints
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,6 +16,9 @@ install:
   - if [ ${USE_CYTHON} ]; then conda install --yes -n mdsa cython; fi
   - source activate mdsa
   - pip install .
+  - pip install coveralls
 script:
-  - nosetests
+  - nosetests --with-coverage --cover-package=mdsa
   - flake8 setup.py mdsa
+after_success:
+  - coveralls

